core/src/banking_stage.rs

```diff
@@ -376,6 +379,20 @@ impl BankingStage {
                     batch_limit,
                     transaction_status_sender.clone(),
                 );
+                let post_buffered_len: usize =
+                    buffered_packets.iter().map(|p| p.1.len()).sum::<usize>();
+                if buffered_len == post_buffered_len
+                    && buffered_process_start.elapsed().as_millis() > 1000
+                {
+                    // Escape hatch if there doesn't seem to be
+                    // progress, maybe receiver has dropped and we need to exit the thread.
```

Comment:
hmmm which receiver would have to be dropped in order for no progress to be made?